ReadingList/ReadingListSQLStorage.swift

```diff
@@ -94,6 +95,13 @@ class ReadingListSQLStorage: ReadingListStorage {
                 return Result(failure: ReadingListStorageError("Can't get first item from results"))
             }
         } else {
+            if let reason = statement.reason {
+                // TODO: this is ugly string matching, but checking for error SQLITE_CONSTRAINT does not say which constraint was violated, and even with checking for SQLITE_CONSTRAINT_UNIQUE extended result code, it is not possible to get which column's uniqueness was violated; so no idea at an "exact" way currently
```

Comment:
The only unique field in the reading list database is the URL. So if you can check for `SQLITE_CONSTRAINT` then that is enough to know that the page was already in the reading list.